### PR TITLE
Fix struct alignments that are not correctly recognized by bindgen

### PR DESCRIPTION
**Pull request by dvic**
@kpp @Fraser999 I've applied the comments from #303 and put them into a new PR. This now includes patching `crypto_onetimeauth_poly1305_state` and adding the tests to `libsodium-sys`. 

Anything I missed 😊?

### CODE REVIEW

File: libsodium-sys/src/gen.sh [hunk 1]
```diff
@@ -9,3 +9,11 @@ bindgen PATH_TO/libsodium-1.0.16/src/libsodium/include/sodium.h -o sodium_bindin
...
   --whitelist-type=$REGEX \
   --whitelist-var=$REGEX
+
+# bindgen fails to compute the alignment in some cases:
+# - the alignment of crypto_onetimeauth_poly1305_state should be 16
+#   see https://github.com/jedisct1/libsodium/blob/1.0.16/src/libsodium/include/sodium/crypto_onetimeauth_poly1305.h#L19
+# - the alignment of crypto_generichash_blake2b_state should be 64
+#   see https://github.com/jedisct1/libsodium/blob/1.0.16/src/libsodium/include/sodium/crypto_generichash_blake2b.h#L23
+# the patch below fixes this
+git apply alignment_fix.patch
```
Comment (kpp) [inline libsodium-sys/src/gen.sh]: new line

**Comment by kpp** [comment]
LGTM.

File: libsodium-sys/src/gen.sh (hunk 1 above)
Comment (dvic) [reply libsodium-sys/src/gen.sh]: Fixed

**Comment by Fraser999** [comment]
Looks good to me too @dvic - thanks! 👍

**Comment by Dylan-DPC** [comment]
bors: r+

**Comment by Dylan-DPC** [comment]
Thanks @dvic

**Pull request merged**